File: 2455-average-value-of-even-numbers-that-are-divisible-by-three/2455-average-value-of-even-numbers-that-are-divisible-by-three.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var averageValue = function (nums) {
    const evenNums = nums.filter(n => n % 2 === 0);
    const divByThree = evenNums.filter(e => e % 3 === 0);
    if (divByThree.length > 0) {
        let sum = 0;
        for (let num of divByThree) {
            sum += num
        }
        return Math.floor(sum / divByThree.length)
    }else{
        return 0
    }
};

File: 2455-average-value-of-even-numbers-that-are-divisible-by-three/2455-average-value-of-even-numbers-that-are-divisible-by-three.cs
public class Solution {
    public int AverageValue(int[] nums) {
        int sum = 0;
        int count = 0;
        foreach(var i in nums)
        {
            if(i % 6 == 0){
                sum += i;
                count++;
            }
        }
        return count == 0 ? 0 : sum/count;
    }
}